crates/deltalake-core/src/kernel/snapshot/log_segment.rs

```diff
@@ -148,15 +149,21 @@ impl LogSegment {
         table_root: &Path,
         start_version: i64,
         end_version: Option<i64>,
-        store: &dyn ObjectStore,
+        log_store: &dyn LogStore,
     ) -> DeltaResult<Self> {
         debug!(
             "try_new_slice: start_version: {}, end_version: {:?}",
             start_version, end_version
         );
+        let max_version_2 = log_store.get_latest_version(start_version).await?;
```

Comment:
should we add a TODO here o.a. saying why this statement exists and the we want to get rid of it?